eth/protocols/snap/handler.go

```diff
@@ -524,13 +520,30 @@ func ServiceGetTrieNodesQuery(chain *core.BlockChain, req *GetTrieNodesPacket, s
 			bytes += uint64(len(blob))
 
 		default:
+			var stRoot common.Hash
 			// Storage slots requested, open the storage trie and retrieve from there
-			account, err := snap.Account(common.BytesToHash(pathset[0]))
-			loads++ // always account database reads, even for failures
-			if err != nil || account == nil {
-				break
+			if snap == nil {
+				var account types.StateAccount
+				// We don't have the requested state snapshotted yet (or it is stale).
+				// We can look  up the account from the account trie instead.
+				blob, resolved, err := accTrie.TryGetNode(pathset[0])
```

Comment:
@rjl493456442 lgty?

Reply:
I think it's not correct. 

`pathset[0]` represents the hash of account address. `TryGetNode` accepts the COMPACT-format node path, but the `pathset[0]` is actually the raw trie node path.

we should use this code instead

```go
account, err := accTrie.TryGetAccountWithPreHashedKey(pathset[0])
if err != nil || account == nil {
	break
}
```

Reply:
Another issue is the counter for resolved nodes. Previously it's 1 for a single database lookup(snapshot), but it's not the case here.

Reply:
The reason I went with `TryGetNode` in this case, is because it's what is used at line 514 -- which I _think_ is totally equivalent "path-wise". So now I'm confused... 

> Another issue is the counter for resolved nodes.

What's the issue? We account for it, just like we do on other lookups.

Reply:
Ah I get it, you mean if `TryGetAccountWithPreHashedKey` is used.